src/overlays/actors/ovl_Obj_Milk_Bin/z_obj_milk_bin.c

```diff
@@ -21,12 +26,80 @@ const ActorInit Obj_Milk_Bin_InitVars = {
     (ActorFunc)ObjMilkBin_Update,
     (ActorFunc)ObjMilkBin_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Milk_Bin_0x80C08E40/ObjMilkBin_Init.asm")
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_HARD,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    { 12, 30, 0, { 0, 0, 0 } },
+};
+
+void ObjMilkBin_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjMilkBin* this = THIS;
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+
+    this->actor.shape.yOffset = 1100.0f;
+    this->disableDraw = 0;
+    this->type = thisx->params;
+
+    if ((this->type == OBJ_MILK_BIN_TYPE_2) && !(gSaveContext.perm.weekEventReg[0x34] & 1)) {
+        this->disableDraw |= 1;
+    }
+}
+
+void ObjMilkBin_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjMilkBin* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void ObjMilkBin_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    ObjMilkBin* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    if (this->type == OBJ_MILK_BIN_TYPE_1) {
+        if (gSaveContext.perm.weekEventReg[0x16] & 1) {
+            if (((gSaveContext.perm.day == 2) && (gSaveContext.perm.isNight == 1)) || (gSaveContext.perm.day >= 3)) {
+                Actor_MarkForDeath(&this->actor);
+                return;
+            }
+        }
+    } else if (this->type == OBJ_MILK_BIN_TYPE_2) {
+        if (gSaveContext.perm.weekEventReg[0x34] & 1) {
+            this->disableDraw &= ~1;
+        } else {
+            this->disableDraw |= 1;
+        }
+    }
+
+    if (!(this->disableDraw & 1)) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Milk_Bin_0x80C08E40/ObjMilkBin_Destroy.asm")
+// gMilkBinMilkJarDL
+extern Gfx D_060004B0[];
```

Comment:
Really small nit: I would put any declarations like this at the top of the file, before any of the definitions?

Reply:
Done :D